common/src/main/java/dev/latvian/mods/kubejs/block/custom/BasicBlockJS.java

```diff
@@ -126,7 +133,13 @@ public MutableComponent getName() {
 	@Override
 	@Deprecated
 	public VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context) {
-		return shape;
+		Map<String, Object> blockPropertyValues = new HashMap<>();
+		state.getProperties().forEach((property) -> {
+			blockPropertyValues.put(property.getName(), state.getValue(property));
+		});
+		var voxelShape = shapeMap.get(blockPropertyValues);
+		if(voxelShape == null || voxelShape.isEmpty()) return shape;
```

Comment:
Ya know, I was doing some testing, and i got a crash because it was null, I think the only reason it can be null, is a different issue, that I am working on, but getOrDefault will prevent a crash if something goes wrong.